app_dart/lib/src/request_handlers/update_agent_health.dart

```diff
@@ -45,13 +50,38 @@ class UpdateAgentHealth extends ApiRequestHandler<UpdateAgentHealthResponse> {
         throw BadRequestException('Invalid agent ID: $agentId');
       },
     );
+    final TabledataResourceApi tabledataResourceApi = await config.createTabledataResourceApi();
+    final List<Map<String, Object>> tableDataInsertAllRequestRows = <Map<String, Object>>[];
 
     agent.isHealthy = isHealthy;
     agent.healthDetails = healthDetails;
     agent.healthCheckTimestamp = DateTime.now().millisecondsSinceEpoch;
 
     await datastore.db.commit(inserts: <Agent>[agent]);
 
+    /// Insert data to [BigQuery] whenever updating data in [Datastore] 
+    tableDataInsertAllRequestRows.add(<String, Object>{
+      'json': <String, Object>{
+        'Timestamp': agent.healthCheckTimestamp,
+        'AgentID': agentId,
+        // TODO(keyonghan): add more detailed states https://github.com/flutter/flutter/issues/44213
+        'Status': isHealthy?'Healthy':'Unhealthy',
+        'Detail': healthDetails,
+      },
+    });
+
+    /// Final [rows] to be inserted to [BigQuery]
+    final TableDataInsertAllRequest rows =
+      TableDataInsertAllRequest.fromJson(<String, Object>{
+      'rows': tableDataInsertAllRequestRows
+    });
+
+    try {
+      await tabledataResourceApi.insertAll(rows, projectId, dataset, table);
+    } catch(ApiRequestError){
+      log.warning('Failed to add $agentId status to BigQuery: $ApiRequestError');
+    }
+
     return UpdateAgentHealthResponse(agent);
   }
 }
```

Comment:
I would put the new changes into a sub-function to avoid interleave the the old logic.

Even better, put the old logic into a sub-function too. Become something like
```
post() {
  _updateDatastore();
  _updateBigQuery();
}
```

Reply:
Added _updateBigquery() as recommended.
As for _updateDatastore(), since we have only one datastore insert script, so just leave it there.